src/absil/ilwrite.fs

```diff
@@ -4462,105 +4474,146 @@ let writeBinaryAndReportMappings (outfile, ilg, pdbfile: string option, signer:
                  b0 reloc2; b1 reloc2; |]
           writePadding os "end of .reloc" (imageEndSectionPhysLoc - relocSectionPhysLoc - relocSectionSize)
 
-          os.Close()
-          
-          try 
-              FileSystemUtilites.setExecutablePermission outfile
-          with _ -> 
-              ()
           pdbData,debugDirectoryChunk,debugDataChunk,textV2P,mappings
-          
-        // Looks like a finally...
-        with e ->   
-            (try 
-                os.Close() 
-                FileSystem.FileDelete outfile 
-             with _ -> ()) 
-            reraise()
-   
-
-    reportTime showTimes "Writing Image"
-     
-    if dumpDebugInfo then 
-        DumpDebugInfo outfile pdbData
+
+    let writeDumpDebugInfo pdbData dumpTo =
+        reportTime showTimes "Dump Debug Info"
+        let dump stream = DumpDebugInfo stream pdbData
```

Comment:
Please avoid little nested helper functions like this - inlining the call to DumpDebugInfo in two places is clearer to read than a nested inner function just called ``dump``.